spec/service/integration/person_intergration_spec.rb

```diff
@@ -0,0 +1,365 @@
+        get "/1/Person/#{person.id}",
+        nil,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        expect(last_response.status).to eq 200
+      end
+
+      it"returns the correct message when successful" do
+        get "/1/Person/#{person.id}",
+        nil,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        message =%({"name": "#{person.name}", "kind": "Person"})
+        expect(last_response.body).to be_json_eql(message)
+      end
+    end
+
+    context "Person doesn't exist" do
+      let(:id) { Hoodoo::UUID.generate}
+
+      it "returns 404" do
+        get "/1/Person/#{id}",
+        nil,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+        expect(last_response.status).to eq 404
+      end
+    end
+  end
+
+  context "#list" do
+    # Method check the http code is correct,
+    # encodes the search query(if any),
+    # and parses the response body so that the body contents may be passed to the compare_lists method.
+    def do_list( search = {}, expected_code = 200 )
+      query = ''
+      unless search.empty?
+        encoded_search = URI.encode_www_form( search )
+        query = '?' << URI.encode_www_form( 'search' => encoded_search )
+      end
+      response = get(
+        "/1/Person#{ query }",
+        nil,
+        { 'CONTENT_TYPE' => 'application/json; charset=utf-8' }
+      )
+      expect( last_response.status).to(
+        eq( expected_code )
+      )
+      return JSON.parse( last_response.body )
+    end
+
+    # compares resource being returned in the service response agains the models created by FactoryBot that are expected to be found in the services response.
+    def compare_lists( resources, *models )
+      models.each_with_index do | model, index |
+        resource = resources[ '_data' ][ index ]
+        expect( model.name                  ).to eq( resource[ 'name'          ] )
+        expect( model.date_of_birth.iso8601 ).to eq( resource[ 'date_of_birth' ] )
+      end
+    end
+
+    context "list when database does not contain any entries" do
+      it "returns that the last_response body is empty" do
+        res = do_list('')
+        compare_lists(res)
+        expect(last_response.body).to have_json_size(0).at_path("_data")
+      end
+    end
+
+    context "Database contains many entries" do
+      before :each do
+        @p1 = FactoryBot.create( :person, :name => 'John 1',    :date_of_birth => '2002-03-01' )
+        @p2 = FactoryBot.create( :person, :name => 'John 2',    :date_of_birth => '2002-09-04' )
+        @p3 = FactoryBot.create( :person, :name => 'Dylan 1',   :date_of_birth => '2000-11-23' )
+        @p4 = FactoryBot.create( :person, :name => 'Dylan 2',   :date_of_birth => '1996-02-01' )
+        @p5 = FactoryBot.create( :person, :name => 'Dylan 3',   :date_of_birth => '1996-02-01' )
+      end
+
+      it "lists all entries" do
+        res = do_list('')
+        compare_lists(res, @p5, @p4, @p3, @p2, @p1)
```

Comment:
Add a `sort` order on the `PersonInterface` to make ordering explicit - see https://github.com/LoyaltyNZ/hoodoo-sample/pull/6/files